solutions/usaco-265.mdx

```diff
@@ -0,0 +1,109 @@
+---
+id: usaco-265
+source: Old Gold
+title: The Cow Run
+author: Ryan Chou
+---
+ 
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_cowrun_mar13.html)
+
+## Explanation
+
+Since we'll always want to pick up a cow when we can, the list of cows that we've visited will form a subarray in the sorted list of cows.
+
+So, we'll sort the cows by their position.
+
+However, we can't always just go from one side to another. In some cases, it might be more optimal to go back and forth from one side to another.
+
+Let's use a range DP.
+#### States
+
+We have to keep track of the cows that we've used. Since this'll always be a range, we'll represent it as $i$ and $j$ (the endpoints of the interval).
+
+We also have to keep track of which side we're on (the left or right).
+
+$\texttt{dp}[i][j][k] = \hspace{0.1cm}$minimum cost to visit all of these cows. 
```

Comment:
maybe explain all the variables right here

Reply:
don't i?

ill change it for $k$

Reply:
oh yeah i'm dumb